src/components/input-field/input-field.tsx

```diff
@@ -462,7 +473,37 @@ export class InputField {
             );
         }
 
-        if (trailingIcon) {
+        if (
+            !this.isInvalid() &&
+            this.showLink &&
+            ['email', 'tel', 'url'].includes(this.type)
+        ) {
+            let link: string;
+            const props: { target?: string } = {};
+            switch (this.type) {
+                case 'email':
+                    link = `mailto:${this.value}`;
+                    break;
+                case 'tel':
+                    link = `tel:${this.value}`;
+                    break;
+                default:
+                    link = `${this.value}`;
+                    props.target = '_blank';
+            }
+
+            html.push(
+                <a
+                    href={link}
+                    class="mdc-text-field__icon"
+                    tabindex={this.disabled ? '-1' : '0'}
+                    role="button"
+                    {...props}
+                >
+                    <limel-icon name={trailingIcon} />
+                </a>
+            );
+        } else if (trailingIcon) {
```

Comment:
I think there may be room for some refactoring to make this nicer, but it's what I have for tonight 😅

Reply:
Yeah both `if`/`else` blocks can probably be extracted to something like
```ts
if (!this.isInvalid() && this.hasLink()) {
    html.push(this.renderLinkIcon());
} else if (trailingIcon) {
    html.push(this.renderTrailingIcon());
}
```

Reply:
⚡